src/workerd/io/io-context.h

```diff
@@ -352,6 +352,8 @@ class IoContext final: public kj::Refcounted, private kj::TaskSet::ErrorHandler
   //
   // Use the ...IoOwn() overload if you need to store this promise in a JS API object.
 
+  bool isOutputGateBroken();
```

Comment:
Non-blocking for this PR but the "output gate" terminology here may not be obvious for folks coming into the codebase fresh. Some additional doc comments in here explaining what this is would likely be helpful.